zcash_client_sqlite/src/lib.rs

```diff
@@ -1106,15 +1125,17 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                                 UnifiedAddress::from_receivers(
                                     Some(output.note().recipient()),
                                     None,
-                                    None
-                                ).expect("UA has an Orchard receiver by construction."),
-                                PoolType::Shielded(ShieldedProtocol::Orchard)
+                                    None,
+                                )
+                                .expect("UA has an Orchard receiver by construction."),
+                                PoolType::Shielded(ShieldedProtocol::Orchard),
                             )
                         } else {
-                            Recipient::InternalAccount(
-                                *output.account(),
-                                Note::Orchard(*output.note()),
-                            )
+                            Recipient::InternalAccount {
+                                receiving_account: *output.account(),
+                                external_address: None,
```

Comment:
Ditto: how do we ensure this is set correctly for cross-account transfers?

Reply:
Same, this is down to `TransferType`